src/App.tsx

```diff
@@ -1,109 +1,39 @@
 import React from 'react';
-
 import './App.scss';
-
-// import postsFromServer from './api/posts';
-// import commentsFromServer from './api/comments';
-// import usersFromServer from './api/users';
+import { PostList } from './components/PostList';
+import { Post } from './types/Post';
+import postsFromServer from './api/posts';
+import commentsFromServer from './api/comments';
+import usersFromServer from './api/users';
+import { User } from './types/User';
+import { Comment } from './types/Comment';
+
+function getUserById(users: User[], postId: number): User | undefined {
+  return users.find(user => (
+    user.id === postId
+  ));
+}
+
+function getCommentsForPost(
+  comments: Comment[], postId: number,
+): Comment[] {
+  return comments.filter(comment => (
+    comment.postId === postId
+  ));
+}
+
+const predaredPosts: Post[] = postsFromServer.map(
+  post => ({
+    ...post,
+    user: getUserById(usersFromServer, post.userId) || null,
```

Comment:
handle it in the `getUserById` function
```suggestion
    user: getUserById(usersFromServer, post.userId),
```